WMF Framework/Remote Notifications/Operations/RemoteNotificationsOperation.swift

```diff
@@ -0,0 +1,16 @@
+class RemoteNotificationsOperation: AsyncOperation {
+    let apiController: RemoteNotificationsAPIController
+    let modelController: RemoteNotificationsModelController
+    let managedObjectContext: NSManagedObjectContext
+
+    init(with apiController: RemoteNotificationsAPIController, modelController: RemoteNotificationsModelController) {
+        self.apiController = apiController
+        self.modelController = modelController
+        self.managedObjectContext = modelController.managedObjectContext
+        super.init()
+    }
+
+    override var shouldExecute: Bool {
+        return WMFAuthenticationManager.sharedInstance.isLoggedIn
```

Comment:
Could this be achieved by stopping the controller when the user isn't logged in? That way the operations are never created in first place, and it eliminates the need for the `shouldExecute` shortcut in `AsyncOperation`